### ISSUE TITLE
Type generation malforming objects with number-like keys

### ISSUE THREAD

**Issue by DamianOsipiuk** [horiuchi/dtsgenerator#523]
Running generator on schema that has number-like keys in objects, prepends `$` sign to every key.
This makes the generated types invalid.
If this is an actual bug, and you do not want to make a breaking change, at least an config option to fix this behavior would be neat.

Example schema:
```
{
  "components": {
    "examples": {},
    "headers": {},
    "parameters": {},
    "requestBodies": {},
    "responses": {},
    "schemas": {
      "MyObj": {
        "properties": {
          "1": {
            "type": "string"
          },
          "2": {
            "type": "string"
          },
          "3": {
            "type": "string"
          },
          "4": {
            "type": "string"
          },
          "5": {
            "type": "string"
          }
        },
        "required": ["5", "4", "3", "2", "1"],
        "type": "object"
      }
    },
    "securitySchemes": {}
  },
  "info": {
    "title": ""
  },
  "openapi": "3.0.0",
  "paths": {
    "/getMyObj": {
      "get": {
        "operationId": "GetMyObj",
        "responses": {
          "200": {
            "description": "Ok",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/MyObj"
                }
              }
            }
          }
        },
        "security": []
      }
    }
  }
}

```

Output:
```
declare namespace Components {
    namespace Schemas {
        export interface MyObj {
            $1: string;
            $2: string;
            $3: string;
            $4: string;
            $5: string;
        }
    }
}
declare namespace Paths {
    namespace GetMyObj {
        namespace Responses {
            export type $200 = Components.Schemas.MyObj;
        }
    }
}
```

Expected output:
```
declare namespace Components {
    namespace Schemas {
        export interface MyObj {
            '1': string;
            '2': string;
            '3': string;
            '4': string;
            '5': string;
        }
    }
}
declare namespace Paths {
    namespace GetMyObj {
        namespace Responses {
            export type $200 = Components.Schemas.MyObj;
        }
    }
}

```

**Comment by DamianOsipiuk** [comment]
It's related to this line https://github.com/horiuchi/dtsgenerator/blob/master/src/core/astBuilder.ts#L12
When i remove the mentioned condition, proper types are generated, while still keeping `types` and `interfaces` prefixed with `$` if they start with number.
There might be some corner cases that I'm not able to see, but this seems like base for the solution.

**Comment by horiuchi** [comment]
I mistakenly thought that all identifiers could not always start with a number.
As for property identifiers, since numbers are also used in Array, we can be defined with words starting with numbers.
I need to sort out which character types can be used for each identifier.

**Comment by horiuchi** [comment]
The definition of the identifier was a bellow page, but the key of the property was defined as a string literal, so it was different.
https://tc39.es/ecma262/#sec-names-and-keywords

Also, I noticed that the current implementation of identifiers does not check for reserved words.

**Comment by horiuchi** [comment]
However, in the current implementation, identifiers do not conflict with reserved words because the first letter is converted to uppercase.

**Comment by horiuchi** [comment]
Released by v3.13.2
